Code/GraphMol/FileParsers/MultithreadedMolSupplier.h

```diff
@@ -85,7 +85,8 @@ class RDKIT_FILEPARSERS_EXPORT MultithreadedMolSupplier : public MolSupplier {
   std::thread d_readerThread;                    //! single reader thread
 
  protected:
-  unsigned int d_lastRecordId = 0;           //! stores last extracted record id
+  std::atomic<unsigned int> d_lastRecordId =
+      0;                                     //! stores last extracted record id
   std::string d_lastItemText;                //! stores last extracted record
```

Comment:
d_lastItemText should be an atomic, but I don't think that std::strings are trivially copyable.

Do we really need the lastItemText and lastRecordId for anything besides debugging?

Reply:
It might be better to attached the recordId as a computed prop to the returned molecule which would remove this issue.